crates/transport/src/value.rs

```diff
@@ -261,6 +261,15 @@ impl_deferred_sync!(CoreVecDecoder<Leb128DecoderI128>);
 impl_deferred_sync!(CoreVecDecoder<Leb128DecoderU128>);
 impl_deferred_sync!(CoreVecDecoder<UnitCodec>);
 
+impl<T, W> Deferred<T> for CoreVecDecoder<OptionDecoder<W>>
```

Comment:
This is not quite right, since it would discard the async value for all `option<stream<T>>`, `option<future<T>>` or nested async values (e.g. `option` of a record, which contains a field of `future<T>` type)